src/boundThreepids.js

```diff
@@ -16,35 +16,41 @@ limitations under the License.
 
 import IdentityAuthClient from './IdentityAuthClient';
 
-export async function getThreepidBindStatus(client, filterMedium) {
+export async function getThreepidsWithBindStatus(client, filterMedium) {
     const userId = client.getUserId();
 
     let { threepids } = await client.getThreePids();
     if (filterMedium) {
         threepids = threepids.filter((a) => a.medium === filterMedium);
     }
 
-    if (threepids.length > 0) {
-        // TODO: Handle terms agreement
-        // See https://github.com/vector-im/riot-web/issues/10522
-        const authClient = new IdentityAuthClient();
-        const identityAccessToken = await authClient.getAccessToken();
-
-        // Restructure for lookup query
-        const query = threepids.map(({ medium, address }) => [medium, address]);
-        const lookupResults = await client.bulkLookupThreePids(query, identityAccessToken);
-
-        // Record which are already bound
-        for (const [medium, address, mxid] of lookupResults.threepids) {
-            if (mxid !== userId) {
-                continue;
+    // Check bind status assuming we have an IS and terms are agreed
+    if (threepids.length > 0 && client.getIdentityServerUrl()) {
```

Comment:
nit (and arguable code style argument): `!!` strings to reinforce that you do actually want the boolean value and haven't forgotten some code